bigbluebutton-html5/imports/ui/components/join-handler/custom-users-settings/component.tsx

```diff
@@ -45,44 +43,55 @@ const CustomUsersSettings: React.FC<CustomUsersSettingsProps> = ({
       return undefined;
     }
 
-    BBBWeb.index(controller.signal)
-      .then(({ data }) => {
-        const url = new URL(`${data.graphqlApiUrl}/userMetadata`);
-        fetch(url, {
-          method: 'get',
+    BBBWeb.request(
+      // Dynamic URL from BBB index info
+      (data) => `${data.graphqlApiUrl}/userMetadata`,
+      {
+        fetchOptions: {
+          method: 'GET',
           credentials: 'include',
           headers: {
             'x-session-token': sessionToken ?? '',
           },
-          signal: controller.signal,
-        })
-          .then((resp) => resp.json())
-          .then((data: Response) => {
-            const filteredData = data.user_metadata.map((uc) => {
-              const { parameter, value } = uc;
-              let parsedValue: string | boolean | string[] = '';
-              try {
-                parsedValue = JSON.parse(uc.value);
-              } catch {
-                parsedValue = value;
-              }
-              return { [parameter]: parsedValue };
-            });
-            setUserSettings(filteredData.reduce((acc, item) => Object.assign(acc, item), {}));
-            setFetched(true);
-            if (timeoutRef.current) {
-              clearTimeout(timeoutRef.current);
-            }
-          }).catch(() => {
-            setError('Error fetching user custom settings');
-            Session.setItem('errorMessageDescription', 'meeting_ended');
-          })
-          .finally(() => {
-            setLoading(false);
-          });
-      }).catch((error) => {
+        },
+        signal: controller.signal,
+        timeout: CONNECTION_TIMEOUT / 3,
```

Comment:
Maybe `20` is too short since it was previously `60`.
`30` might be a more reasonable timeout.